Carlos/Composed.swift

```diff
@@ -49,8 +49,23 @@ extension CacheLevel {
         return request.future
       },
       setClosure: { (value, key) in
+        let promise = Promise<()>()
+
         self.set(value, forKey: key)
-        cache.set(value, forKey: key)
+          .onSuccess { _ in
```

Comment:
I think we can borrow an idea in CloudKit where they use a concept of partial failure. So `promise.fail` would be called but with an an error that describes that only one of the writes failed. Maybe even trying to clarify which exact cache error failed if possible.

Unfortunately, The errors in Carlos as I see are a bit hairy, maybe even limited by a current Swift implementation details. It surprised me that a generic class can't have nested enums, thus I wasn't able to namespace a new enum describing `set` errors. I also wasn't able to come up with a good name for that enum, as Carlos already has `FetchError`, but is `SetError` better than `WriteError`? And what if you have a network cache layer, does it make sense to "write" to network? 

Because of all this, I would refrain from implementing this error handling by myself for now. But I agree this needs a `TODO` or `FIXME` note there.